src/azul/plugins/__init__.py

```diff
@@ -700,6 +707,21 @@ def fetch_bundle(self, bundle_fqid: BUNDLE_FQID) -> BUNDLE:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def list_files(self, source: SOURCE_REF, prefix: str) -> list['File']:
+        """
+        List the files in the given source whose UUID starts with the given
+        prefix
```

Comment:
```suggestion
        List the files in the given source whose UUID starts with the given
        prefix.
```

This should explain the relationship between the `prefix` argument and the `prefix` attribute of the source. If there is a constraint, like that one is a prefix of the other, that should also be required (using `R()`), if it isn't already.

Reply:
Agreed at PL to add prefix validation to both this method and `list_bundles`, analogous to the existing `_assert_source` validator method.

Reply:
https://github.com/DataBiosphere/azul/issues/7040